example/test/react_test_components.dart

```diff
@@ -223,23 +223,168 @@ class _ContextConsumerComponent extends react.Component {
       context.toString(),
     ]);
   }
 }
 
-var grandchildContextConsumerComponent = react.registerComponent(() => new _GrandchildContextConsumerComponent());
+var grandchildLegacyContextConsumerComponent =
+    react.registerComponent(() => new _GrandchildLegacyContextConsumerComponent());
+
+////
+// REACT NEW CONTEXT COMPONENTS
+////
+class _NewContextRefComponent extends react.Component2 {
+  render() {
+    return react.div({}, props['children']);
+  }
+
+  test() {
+    print('test');
+  }
+}
+
+var newContextRefComponent = react.registerComponent(() => new _NewContextRefComponent());
+
+int calculateChangedBits(currentValue, nextValue) {
+  int result = 1 << 1;
+  if (nextValue['renderCount'] % 2 == 0) {
+    result |= 1 << 2;
+  }
+  return result;
+}
+
+var TestNewContext = createContext({'renderCount': 0}, calculateChangedBits);
+
+class _NewContextProviderComponent extends react.Component2 {
+  _NewContextRefComponent componentRef;
+
+  getInitialState() => {'renderCount': 0, 'complexMap': false};
+
+  printMe() {
+    print('printMe!');
+  }
+
+  render() {
+    Map provideMap = {'renderCount': this.state['renderCount']};
+
+    Map complexValues = {
+      'callback': printMe,
+      'dartComponent': newContextRefComponent,
+      'map': {
+        'bool': true,
+        'anotherDartComponent': newContextRefComponent,
+      },
+      'componentRef': componentRef,
+    };
+
+    if (state['complexMap']) {
+      provideMap.addAll(complexValues);
+    }
+
+    Map newContextRefComponentProps = {
+      'key': 'ref2',
+      'ref': (ref) {
+        componentRef = ref;
+      }
+    };
+
+    return react.ul({
+      'key': 'ulasda',
+    }, [
+      newContextRefComponent(newContextRefComponentProps, []),
+      react.button({
+        'type': 'button',
+        'key': 'button',
+        'className': 'btn btn-primary',
+        'onClick': _onButtonClick,
+      }, 'Redraw'),
+      react.button({
+        'type': 'button',
+        'key': 'button_complex',
+        'className': 'btn btn-primary',
+        'onClick': _onComplexClick,
+      }, 'Redraw With Complex Value'),
+      react.br({'key': 'break1'}),
+      'TestContext.Provider props.value: ${provideMap}',
+      react.br({'key': 'break2'}),
+      react.br({'key': 'break3'}),
+      TestNewContext.Provider(
+        {'key': 'tcp', 'value': provideMap},
+        props['children'],
+      ),
+    ]);
+  }
+
+  _onComplexClick(event) {
+    this.setState({'complexMap': true, 'renderCount': this.state['renderCount'] + 1});
+  }
+
+  _onButtonClick(event) {
+    this.setState({'renderCount': this.state['renderCount'] + 1, 'complexMap': false});
+  }
+}
+
+var newContextProviderComponent = react.registerComponent(() => new _NewContextProviderComponent());
+
+class _NewContextConsumerComponent extends react.Component2 {
+  render() {
+    return TestNewContext.Consumer({'unstable_observedBits': props['unstable_observedBits']}, (value) {
```

Comment:
Why are we using `unstable` ReactJS APIs in something extending from `Component2`?

Reply:
This is just an example but, the `unstable_observedBits` prop functions better than the stable version of the prop: `observedBits`... thats why its used here

Reply:
Sorry for the delayed reply.

I get that its just an example... but it is an example intended to exercise the functionality in-lieu of a formal functional test - which is why I'm concerned about our implementation possibly not functioning well with the stable version of `observedBits`.